pkg/services/rendering/http_mode.go

```diff
@@ -36,109 +36,112 @@ var (
 	remoteVersionRefreshInterval               = time.Minute * 15
 )
 
+// renderViaHTTP renders PNG or PDF via HTTP
 func (rs *RenderingService) renderViaHTTP(ctx context.Context, renderType RenderType, renderKey string, opts Opts) (*RenderResult, error) {
```

Comment:
This is the biggest change `renderViaHTTP` and `renderCSVViaHTTP` were almost identical. I've extracted all duplicated code. They could be the same function but it will be a bigger change because of returning and parameters types are different.